### PR TITLE
[Core] Clean up legacy meshid from processes

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Removing legacy  `mesh_id`. Draft until see that all tests pass.

**🆕 Changelog**

- [Clean up legacy `mesh_id` from processes](https://github.com/KratosMultiphysics/Kratos/commit/2fb4edd6cce0607bff4cb4098fc7be1788fd9076)

### CODE REVIEW

File: kratos/python_scripts/assign_scalar_variable_process.py [hunk 1]
```diff
@@ -74,7 +72,7 @@ def __init__(self, Model, settings ):
             raise Exception(msg)
 
         self.model_part = Model[settings["model_part_name"].GetString()]
-        self.mesh = self.model_part.GetMesh(settings["mesh_id"].GetInt())
```
Comment (rubenzorrilla) [inline kratos/python_scripts/assign_scalar_variable_process.py:77]: I think we can even remove `self.mesh` if we pass the nodes container as `self.model_part.Nodes` below.

**Comment by loumalouomega** [comment]
`mesh_id` is everywhere

**Comment by philbucher** [comment]
Please add backward compatibility, otherwise this will break lots of cases

**Comment by loumalouomega** [comment]
> Please add backward compatibility, otherwise this will break lots of cases

Done

**Comment by loumalouomega** [comment]
@roigcarlo I think there is a bug, CI is not executing at all (maybe exceptions are not well defined for doc)

**Comment by roigcarlo** [comment]
Mm which CI is not running?

**Comment by loumalouomega** [comment]
> Mm which CI is not running?

Okay, last one actually worked

**Comment by loumalouomega** [comment]
👋

**Comment by philbucher** [comment]
As much as it is annoying, I recommend reverting some of the updates in the apps to make sure that the deprecation works

This is a necessary yet sensitive change that could break many existing cases

**Comment by rubenzorrilla** [comment]
😿

**Pull request closed without merging**